markov/tmatrix.R

```diff
@@ -1101,24 +1122,160 @@ get_results <- function(bat, pitch, stad, home, temp, date,
   
   matchups <- list()
   
-  for (b in candidate_batters) 
-  {
-    for (p in candidate_pitchers)
-    {
-      for (s in states) 
-      {
+  batter_num <- 1
+  
+  for (b in candidate_batters) {
+    if (length(candidate_batters) > 1) {
+      # Create list of states to iterate over
+      states <- list(State$new(top = away, 
```

Comment:
The next 150 lines is too much code.  Can you make a nested loop which iterates on the different basestates and outs and then only have a single State$new()